specification/cognitiveservices/data-plane/Personalizer/preview/v2022-09-01-preview/Personalizer.json

```diff
@@ -0,0 +1,2847 @@
+			"properties": {
+				"data": {
+					"description": "Time series.",
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/Data"
+					},
+					"readOnly": true
+				}
+			}
+		},
+		"PolicyResult": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"description": "The name of the Learning Settings.",
+					"type": "string",
+					"readOnly": true
+				},
+				"slotId": {
+					"description": "Slot id. Empty for overall results and single slot results.",
+					"type": "string",
+					"readOnly": true
+				},
+				"arguments": {
+					"description": "The arguments of the Learning Settings.",
+					"type": "string",
+					"readOnly": true
+				},
+				"policySource": {
+					"description": "The source of the Learning Settings.",
+					"enum": [
+						"Online",
+						"Baseline",
+						"Random",
+						"Custom",
+						"OfflineExperimentation"
+					],
+					"type": "string",
+					"readOnly": true,
+					"x-ms-enum": {
+						"name": "PolicySource",
+						"modelAsString": true
+					}
+				},
+				"summary": {
+					"allOf": [
+						{
+							"$ref": "#/definitions/Summary"
+						}
+					],
+					"readOnly": true
+				},
+				"timeseries": {
+					"allOf": [
+						{
+							"$ref": "#/definitions/TimeSeries"
+						}
+					],
+					"readOnly": true
+				}
+			}
+		},
+		"EvaluationResult": {
+			"type": "object",
+			"properties": {
+				"startTime": {
+					"format": "date-time",
+					"description": "Start time of the time series.",
+					"type": "string",
+					"readOnly": true
+				},
+				"endTime": {
+					"format": "date-time",
+					"description": "End time of the time series.",
+					"type": "string",
+					"readOnly": true
+				},
+				"interval": {
+					"format": "duration",
+					"description": "Aggregation window.",
+					"type": "string",
+					"readOnly": true
+				},
+				"value": {
+					"description": "Time series of the learning settings results.",
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/PolicyResult"
+					}
+				}
+			}
+		},
+		"Policy": {
+			"type": "object",
+			"properties": {
+				"name": {
+					"description": "The name of the Learning Settings.",
+					"type": "string",
+					"readOnly": true
+				},
+				"arguments": {
+					"description": "The arguments of the Learning Settings.",
+					"type": "string",
+					"readOnly": true
+				},
+				"policySource": {
+					"description": "The source of the Learning Settings.",
+					"enum": [
+						"Online",
+						"Baseline",
+						"Random",
+						"Custom",
+						"OfflineExperimentation"
+					],
+					"type": "string",
+					"readOnly": true,
+					"x-ms-enum": {
+						"name": "PolicySource",
+						"modelAsString": true
+					}
+				}
+			}
+		},
+		"Evaluation": {
+			"description": "A counterfactual evaluation.",
+			"type": "object",
+			"properties": {
+				"id": {
+					"description": "The ID of the evaluation.",
+					"maxLength": 256,
+					"type": "string",
+					"readOnly": true
```

Comment:
Don't mark properties as `readOnly` in schemas that are never part of a request body. This allows you to mark properties as required when they are always present in the response.

Reply:
I am not following the comment. These properties are only present in the response and are marked as readonly because when we generate sdk we do not want to create setters for the response properties. Are you suggesting removing all the readonly properties? @mikekistler

Reply:
Synced offline. We decided to keep the readOnly properties because in java that is the only way to suppress the setters.